All_years/04-Social_Care/Care_Home_data.R

```diff
@@ -0,0 +1,304 @@
+  arrange(sending_location, social_care_id, ch_admission_date, period) %>%
+  group_by(sending_location, social_care_id, ch_admission_date, period) %>%
+  # fill in nursing care provision when missing but present in the following entry
+  fill(nursing_care_provision) %>%
+  # tidy up ch_provider using "6" when disagreeing values
+  mutate(
+    min_ch_provider = min(ch_provider),
+    max_ch_provider = max(ch_provider)
+  ) %>%
+  mutate(ch_provider = if_else(min_ch_provider != max_ch_provider, "6", ch_provider)) %>%
+  select(-c(min_ch_provider, max_ch_provider)) %>%
+  # when multiple social_care_id from sending_location for single CHI
+  # replace social_care_id with latest
+  mutate(latest_sc_id = last(social_care_id)) %>%
+  # count changed social_care_id
+  mutate(
+    changed_sc_id = if_else(!is.na(chi) & social_care_id != latest_sc_id, 1, 0),
+    social_care_id = if_else(!is.na(chi) & social_care_id != latest_sc_id,
+      latest_sc_id, social_care_id
+    )
+  ) %>%
+  # remove any duplicate records
+  distinct() %>%
+  # counter for split episodes
+  mutate(split_episode_counter = pmax(nursing_care_provision != lag(nursing_care_provision), FALSE, na.rm = TRUE)) %>%
+  mutate(sum_split_episode = cumsum(split_episode_counter)) %>%
+  ungroup()
+
+# count changed social_care_id
+ch_data_clean %>% count(changed_sc_id)
+
+
+# Merge Records ---------------------------------------
+
+# to a single row per episode where admission the same
+ch_episode <- ch_data_clean %>%
+  # when nursing_care_provision is different on records within the episode, split the episode at this point
+  group_by(chi, sending_location, social_care_id, ch_admission_date, sum_split_episode, nursing_care_provision) %>%
+  summarise(
+    ch_discharge_date = last(ch_discharge_date),
+    ch_provider = max(ch_provider),
+    record_date = max(record_date),
+    qtr_start = max(qtr_start),
+    sc_latest_submission = max(period),
+    ch_name = last(ch_name),
+    ch_postcode = last(ch_postcode),
+    reason_for_admission = last(reason_for_admission),
+    gender = first(gender),
+    dob = first(dob),
+    postcode = first(postcode)
+  ) %>%
+  ungroup() %>%
+  # Amend dates for split episodes
+  # Change the start and end date as appropriate when an episode is split, using the start / end date of the submission quarter
+  arrange(chi, sending_location, social_care_id, ch_admission_date, sc_latest_submission) %>%
+  group_by(chi, sending_location, social_care_id, ch_admission_date) %>%
+  # counter for latest submission
+  mutate(latest_submission_counter = pmax(sc_latest_submission != lag(sc_latest_submission), FALSE, na.rm = TRUE)) %>%
+  mutate(sum_latest_submission = cumsum(latest_submission_counter)) %>%
+  mutate(
+    ch_admission_date = if_else(sum_latest_submission == min(sum_latest_submission), ch_admission_date, qtr_start),
+    ch_discharge_date = if_else(sum_latest_submission == max(sum_latest_submission), ch_discharge_date, record_date)
+  ) %>%
```

Comment:
Just adding some comments as it took me a while to figure this out! Nice work 😄 
```suggestion
  mutate(
  # If it's the first episode(s) then keep the admission date(s), otherwise use the start of the quarter
    ch_admission_date = if_else(sum_latest_submission == min(sum_latest_submission), ch_admission_date, qtr_start),
      # If it's the last episode(s) then keep the discharge date(s), otherwise use the end of the quarter
    ch_discharge_date = if_else(sum_latest_submission == max(sum_latest_submission), ch_discharge_date, record_date)
  ) %>%
```